src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -975,14 +972,24 @@ This does not pose a problem by itself because they can't be accessed directly."
                         // this doesn't come from a macro that has #[allow_internal_unstable]
                         !self.span.allows_unstable()
                     {
-                        let mut err = self.tcx.sess.struct_span_err(self.span,
-                            &format!("`{}` is not yet stable as a const fn",
-                                     self.tcx.item_path_str(def_id)));
-                        help!(&mut err,
-                              "in Nightly builds, add `#![feature({})]` \
-                               to the crate attributes to enable",
-                              feature_name);
-                        err.emit();
+                        if self.mode == Mode::Fn {
+                            // We are in a normal function
+                            // with a turned off feature gate. We can still call the function
+                            // but we can't promote it
+                            self.qualif = Qualif::NOT_CONST;
```

Comment:
Why is this inside the `if`?  IMO the flags should be as general as possible, so we can make sure less problems show up again. Also, shouldn't this be something like `self.add(Qualif::NOT_CONST)`?

Reply:
Doesn't make much difference as the other branch errors, but seems more consistent indeed